pandas/core/reshape/tile.py

```diff
@@ -391,10 +391,13 @@ def _bins_to_cuts(
             bins = unique_bins
 
     side = "left" if right else "right"
-    ids = ensure_int64(bins.searchsorted(x, side=side))
+    ids = ensure_int64(bins.searchsorted(np.where(~isna(x), x, -1), side=side))
```

Comment:
For `cut` we handle this by converting in the `cut` method:
https://github.com/pandas-dev/pandas/blob/882dbee0b781696db06b2dad00bf3b80eda729f7/pandas/core/reshape/tile.py#L210-L214

It looks like we can do this is a similar place within the `qcut` method, and I think it makes more sense to do that so we avoid doing the `np.where` in cases where it's not needed.

Reply:
Wouldn't this approach also not work for data where -1 is valid?

Reply:
@jschendel Yes, that makes more sense. @WillAyd I think the NA values get masked later so the place holder you use can be arbitrary.